src/components/lab/ElectricFieldVectorGroups.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Group } from 'react-konva';
+import PropTypes from 'prop-types';
+import ElectricFieldVectorGroup from './ElectricFieldVectorGroup';
+import {
+  INFRARED_RADIATION_CURVE_INTERVAL,
+  INFRARED_RADIATION_PERIOD,
+  INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL,
+  SINE_CURVE_AMPLITUDE,
+  Y_SHIFT_PER_INTERVAL,
+} from '../../config/constants';
+
+const ElectricFieldVectorGroups = ({ x }) => {
+  const { height: stageHeight } = useSelector(
+    ({ layout }) => layout.lab.stageDimensions,
+  );
+  const intervalCount = useSelector(({ lab }) => lab.intervalCount);
+
+  return (
+    <Group>
+      <ElectricFieldVectorGroup
+        x={
+          x -
+          SINE_CURVE_AMPLITUDE +
+          SINE_CURVE_AMPLITUDE *
+            Math.sin(
+              (intervalCount -
+                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
+                Y_SHIFT_PER_INTERVAL *
+                INFRARED_RADIATION_PERIOD,
+            )
+        }
+        y={stageHeight - INFRARED_RADIATION_CURVE_INTERVAL}
+        direction={1}
+      />
+      <ElectricFieldVectorGroup
+        x={
+          x +
+          SINE_CURVE_AMPLITUDE -
+          SINE_CURVE_AMPLITUDE *
+            Math.sin(
+              (intervalCount -
+                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
+                Y_SHIFT_PER_INTERVAL *
+                INFRARED_RADIATION_PERIOD,
+            )
+        }
+        y={stageHeight - 3 * INFRARED_RADIATION_CURVE_INTERVAL}
+        direction={-1}
+      />
+      <ElectricFieldVectorGroup
+        x={
+          x -
+          SINE_CURVE_AMPLITUDE +
+          SINE_CURVE_AMPLITUDE *
+            Math.sin(
+              (intervalCount -
+                INTERVALS_TO_COMPLETE_INFRARED_RADIATION_INTERVAL) *
+                Y_SHIFT_PER_INTERVAL *
+                INFRARED_RADIATION_PERIOD,
+            )
+        }
+        y={stageHeight - 5 * INFRARED_RADIATION_CURVE_INTERVAL}
+        direction={1}
+      />
```

Comment:
Would it be clearer if you used a loop to create a fixed number of groups? I guess it should be related to the frequency of the radiations.

Reply:
Yes, somehow this has to be cleaned up... Let me look at it again. Although it's straightforward to find the y-points and then `.map()` each onto an `<ElectricFieldVectorGroup />`, defining the `x` and `direction` props needs some thinking...